ee/clickhouse/queries/funnels/funnel.py

```diff
@@ -1,7 +1,139 @@
+from typing import List
+
 from ee.clickhouse.queries.funnels.base import ClickhouseFunnelBase
 from ee.clickhouse.sql.funnels.funnel import FUNNEL_SQL
 
 
 class ClickhouseFunnel(ClickhouseFunnelBase):
     def get_query(self, format_properties):
         return FUNNEL_SQL.format(**format_properties)
+
+
+class ClickhouseFunnelNew(ClickhouseFunnelBase):
+    def get_query(self, format_properties):
+        return FUNNEL_SQL.format(**format_properties)
+
+    def test_query(self):
+        return self.get_step_counts_query()
+
+    def get_step_counts_query(self):
+
+        formatted_query = ""
+        max_steps = len(self._filter.entities)
+        if max_steps >= 2:
+            formatted_query = self.build_step_subquery(2, max_steps)
+        else:
+            formatted_query = super()._get_inner_event_query()
+
+        breakdown_prop = self._get_breakdown_prop()
+
+        return f"""
+        SELECT {self._get_count_columns(max_steps)}, {self._get_step_time_avgs(max_steps)} {breakdown_prop} FROM (
+            SELECT person_id, {self._get_step_time_avgs(max_steps)}, max(steps) AS furthest {breakdown_prop} FROM (
+                SELECT *, {self._get_sorting_condition(max_steps, max_steps)} AS steps, {self._get_step_times(max_steps)} FROM (
+                    {formatted_query}
+                ) WHERE step_0 = 1
+            ) GROUP BY person_id {breakdown_prop}
+        ) {'GROUP BY prop' if breakdown_prop != '' else ''} SETTINGS allow_experimental_window_functions = 1
+        """
+
+    def build_step_subquery(self, level_index: int, max_steps: int):
+        breakdown_prop = self._get_breakdown_prop()
+        if level_index >= max_steps:
+            return f"""
+            SELECT 
+            person_id,
+            timestamp,
+            {self.get_partition_cols(1, max_steps)}
```

Comment:
If I understand correctly

```suggestion
            {self.get_partition_cols(level_index, max_steps)}
```

Reply:
step_i seems to be 0 indexed, while level_index seems to be 1 indexed? I think easier if both follow the same?